mlapi/holiday_test.go

```diff
@@ -0,0 +1,206 @@
+		))
+		require.NoError(t, err)
+	}))
+	defer s.Close()
+
+	c, err := New(s.URL, Config{})
+	require.NoError(t, err)
+	ctx := context.Background()
+
+	id := "8b154ff8-3d64-4b79-8b26-02b4baeb44e4"
+	holiday, err := c.Holiday(ctx, id)
+	require.NoError(t, err)
+	assert.Equal(t, id, holiday.ID)
+	assert.Equal(t, "Test Holiday", holiday.Name)
+	assert.Equal(t, holiday.CustomPeriods, CustomPeriods([]CustomPeriod{
+		{
+			StartTime: time.Date(2023, time.January, 1, 0, 0, 0, 0, time.UTC),
+			EndTime:   time.Date(2023, time.January, 1, 23, 59, 59, 0, time.UTC),
+		},
+	}))
+	assert.Nil(t, holiday.ICalURL)
+	assert.Nil(t, holiday.ICalTimeZone)
+}
+
+func TestUpdateHoliday(t *testing.T) {
+	id := "8b154ff8-3d64-4b79-8b26-02b4baeb44e4"
+	holiday := Holiday{
+		ID: id,
+	}
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
+			http.Error(w, "method not allowed", http.StatusMethodNotAllowed)
+			return
+		}
+		if r.URL.Path != "/manage/api/v1/holidays/"+id {
+			http.Error(w, "not found", http.StatusNotFound)
+			return
+		}
+		parsedHoliday := Holiday{}
+		dec := json.NewDecoder(r.Body)
+		err := dec.Decode(&parsedHoliday)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		if parsedHoliday.ID != "" {
+			http.Error(w, "id should be empty when updating", http.StatusBadRequest)
+			return
+		}
+		parsedHoliday.ID = id
+		assert.Equal(t, holiday, parsedHoliday)
+		enc := json.NewEncoder(w)
+		_ = enc.Encode(holidayResponseWrapper{Data: parsedHoliday})
+	}))
+	defer s.Close()
+
+	c, err := New(s.URL, Config{})
+	require.NoError(t, err)
+	ctx := context.Background()
+
+	returnedHoliday, err := c.UpdateHoliday(ctx, holiday)
+	require.NoError(t, err)
+	assert.Equal(t, holiday, returnedHoliday)
+}
+
+func TestDeleteHoliday(t *testing.T) {
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "DELETE" {
+			http.Error(w, "method not allowed", http.StatusMethodNotAllowed)
+			return
+		}
+		if r.URL.Path != "/manage/api/v1/holidays/8b154ff8-3d64-4b79-8b26-02b4baeb44e4" {
+			http.Error(w, "not found", http.StatusNotFound)
+			return
+		}
+		_, err := w.Write([]byte("successfully deleted"))
+		require.NoError(t, err)
+	}))
+	defer s.Close()
+
+	c, err := New(s.URL, Config{})
+	require.NoError(t, err)
+	ctx := context.Background()
+
+	err = c.DeleteHoliday(ctx, "8b154ff8-3d64-4b79-8b26-02b4baeb44e4")
+	require.NoError(t, err)
+}
+
+func TestLinkJobHolidays(t *testing.T) {
```

Comment:
This test should also move to `job_test.go` and get renamed to `TestLinkHolidaysToJob`.